### PR TITLE
Add megatron-lm Dist Ckpt -> NeMo Conversion Script

### PR DESCRIPTION
**Pull request by suiyoubi**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by yaoyu-33** [comment]
can you please also add docs under docs checkpoint part?

File: examples/nlp/language_modeling/megatron_dist_ckpt_to_nemo.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+from nemo.collections.nlp.models.language_modeling.megatron_gpt_model import MegatronGPTModel
+from nemo.collections.nlp.modules.common.megatron.megatron_init import initialize_model_parallel_for_nemo
+from nemo.collections.nlp.parts.megatron_trainer_builder import MegatronTrainerBuilder
+from nemo.collections.nlp.parts.nlp_overrides import NLPSaveRestoreConnector
+from nemo.core.config import hydra_runner
+from nemo.utils import AppState, logging
+from nemo.utils.distributed import initialize_distributed
+
+mp.set_start_method("spawn", force=True)
```
Comment (yaoyu-33) [inline examples/nlp/language_modeling/megatron_dist_ckpt_to_nemo.py:13]: this is not needed.

**Comment by eagle705** [comment]
Hi, @suiyoubi @yaoyu-33 
Do we have a plan to merge this PR into main branch?

File: examples/nlp/language_modeling/megatron_dist_ckpt_to_nemo.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+    logging.info("\n\n************** Converted NeMo Configuration ***********")
+    logging.info(f'\n{OmegaConf.to_yaml(cfg)}')
+    return cfg
+
+
+def load_distributed_checkpoint(checkpoint_path, gpt_model):
+    logging.info('Loading Megatron-LM model..')
+    sharded_state_dict = gpt_model.sharded_state_dict(prefix='')
+    checkpoint = dist_checkpointing.load(sharded_state_dict=sharded_state_dict, checkpoint_dir=checkpoint_path)
```
Comment (eagle705) [inline examples/nlp/language_modeling/megatron_dist_ckpt_to_nemo.py:145]: maybe we need to add strict arg
ex)
```
checkpoint = dist_checkpointing.load(sharded_state_dict=sharded_state_dict, checkpoint_dir=checkpoint_path, strict='log_all')
```
